VideoLocker/src/androidTest/java/org/edx/mobile/test/feature/interactor/RegistrationScreenInteractor.java

```diff
@@ -0,0 +1,98 @@
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+public class RegistrationScreenInteractor {
+
+    RegistrationDescription registrationFields;
+
+    public RegistrationScreenInteractor observeRegistrationScreen() {
+        final MainApplication app = MainApplication.instance();
+        try {
+            registrationFields = app.getInjector().getInstance(EdxEnvironment.class).getServiceManager().getRegistrationDescription();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        final CharSequence title = ResourceUtil.getFormattedString(app.getResources(), R.string.register_title, "platform_name", app.getInjector().getInstance(Config.class).getPlatformName());
+        onView(withId(R.id.activity_title)).check(matches(withText(title.toString())));
+        onEmailView().check(matches(isCompletelyDisplayed()));
+        onNameView().check(matches(isCompletelyDisplayed()));
+        onUsernameView().check(matches(isCompletelyDisplayed()));
+        onPasswordView().check(matches(isCompletelyDisplayed()));
+        onCountryView().check(matches(isCompletelyDisplayed()));
+        onCreateAccountButton().check(matches(isCompletelyDisplayed()));
+        return this;
+    }
+
+    public MyCoursesScreenInteractor createAccount(Credentials credentials) {
+        onEmailView().perform(typeText(credentials.email));
+        onNameView().perform(typeText("Test Account"));
+        onUsernameView().perform(typeText(credentials.username));
+        onPasswordView().perform(typeText(credentials.password));
+        onCountryView().perform(SpinnerViewAction.selectSpinnerItem(1));
+        onCreateAccountButton().perform(click());
+        return new MyCoursesScreenInteractor();
+    }
+
+    private ViewInteraction onEmailView() {
+        return onView(withHint(findHintByName("email")));
+    }
+
+    private ViewInteraction onNameView() {
+        return onView(withHint(findHintByName("name")));
+    }
+
+    private ViewInteraction onUsernameView() {
+        return onView(withHint(findHintByName("username")));
+    }
+
+    private ViewInteraction onPasswordView() {
+        return onView(withHint(findHintByName("password")));
+    }
+
+    private ViewInteraction onCountryView() {
+        return onView(withText(findHintByName("country")));
+    }
+
+    private ViewInteraction onCreateAccountButton() {
+        return onView(withId(R.id.create_account_tv));
+    }
+
+    /**
+     * Registration fields are generated at runtime using the
+     * `config/registration_form.json` file. The `label` field
+     * in the json file is translated as the hint for input
+     * views on registration screen.
+     *
+     * @param name Value of the `name` field in json field.
+     * @return Hint of the input view.
+     */
+    private String findHintByName(String name) {
+        for (RegistrationFormField formField : registrationFields.getFields()) {
```

Comment:
Is there any way we can look things up directly by the name instead of having to indirect through the label? I'm not really expecting two things with different names, but the same label, but it'd be nice if we could limit ourselves to views that are definitely just the form field for that name.